SYCL/Matrix/get_coord_bf16_matB_impl.hpp

```diff
@@ -0,0 +1,218 @@
+            i=7, [1, 3]                     i=7, [1, 7]                   i=7, [1, 63]  |
+            ...                             ...                           ....          |
+            i=28,[7, 0]                     i=28,[7, 4]                   i=28,[7, 60]  |
+            i=29,[7, 1]                     i=29,[7, 5]                   i=29,[7, 61]  |
+            i=30,[7, 2]                     i=30,[7, 6]                   i=30,[7, 62]  |
+            i=31,[7, 3]                     i=31,[7, 7]                   i=31,[7, 63]  |
+*/
+// clang-format on
+
+template <typename T, size_t M, size_t N>
+void matrix_sum_cols(queue q, big_matrix<T, M, N> &B, nd_range<2> &r) {
+  buffer<int8_t, 2> bufB(B.get_data(), range<2>(M, N));
+  // size of vector is known because SG size of set by the user in this case
+  int sum_cols[N] = {0};
+  buffer<int> sum_cols_v(sum_cols, N); // there are total of tK/4 * 2, 16 rows
+  q.submit([&](handler &cgh) {
+     auto accB = bufB.get_access<access::mode::read_write>(cgh);
+
+     auto v = sum_cols_v.get_access<access::mode::atomic>(cgh);
+     auto os = sycl::stream(100000, 6144, cgh);
+
+     cgh.parallel_for<class add_matrix>(
+         r, [=](nd_item<2> spmd_item) [[intel::reqd_sub_group_size(SG_SZ)]] {
+           const auto global_idx = spmd_item.get_global_id(0);
+           const auto global_idy = spmd_item.get_global_id(1);
+           const auto sg_startx = global_idx - spmd_item.get_local_id(0);
+           const auto sg_starty = global_idy - spmd_item.get_local_id(1);
+
+           ext::oneapi::sub_group sg = spmd_item.get_sub_group();
+
+           // TK = 32, TN = 16
+           joint_matrix<sub_group, int8_t, use::b, TK, TN,
+                        ext::intel::experimental::matrix::layout::packed>
+               sub_b;
+
+           joint_matrix_load(sg, sub_b,
+                             accB.get_pointer() + (global_idx * (TK / 4) * N) +
+                                 sg_starty / SG_SZ * TN * 4,
+                             N);
+
+           int32_t sum_local_cols[N] = {0}; // 4 local cols, N total
+           // sub_b has 32x16 elements, 32 elements per WI, 4 per WI per row
+           auto wiData =
+               sycl::ext::intel::experimental::matrix::get_wi_data(sg, sub_b);
+
+           size_t
+               global_index; // Index into the result array that holds the sums.
+
+           // Keep track of cols handled in this WI
+           int32_t handled_cols[N] = {-1};
+
+           //  each WI calculates local sum of cols
+           for (int i = 0; i < wiData.length(); ++i) {
+             // get the index of the element in the submatrix
+             auto dataItem = wiData[i];
+             auto [row, col] = dataItem.get_coord();
+
+             // Calculation of global index
+             int sg_idx = (int)global_idy / SG_SZ;
+             global_index = col + sg_idx * 4 /*VNNI_FACTOR*/ * SG_SZ;
```

Comment:
Should this line be `global_index = col * 4 + row % 4 + sg_idx * 4 /*VNNI_FACTOR*/ * SG_SZ;` instead?

`[row, col]` are the coordinates within the non-VNNIed subgroup matrix (32x16). I think we need to map `[row, col]` back to VNNIed indices when summing columns.